src/internal_modules/roc_core/list.h

```diff
@@ -147,6 +147,36 @@ class List : public NonCopyable<> {
         insert_(element, NULL);
     }
 
+    //! Pop first element from list
+    //!
+    //! @remarks
+    //!  - removes first element of list
+    //!  - releases ownership of removed element
+    //!
+    //! @pre
+    //!  the list should not be empty.
+
+    void pop_front() {
+        if (size_ == 0) {
+            roc_panic("List is empty");
+        }
+        remove(*container_of_(head_.next));
+    }
+    //! Pop last element from list
+    //!
+    //! @remarks
+    //!  - removes last element of list
+    //!  - releases ownership of removed element
+    //!
+    //! @pre
+    //!  the list should not be empty.
+
+    void pop_back() {
+        if (size_ == 0) {
+            roc_panic("List is empty");
```

Comment:
We usually format messages as: `component: text`, in lower-case. 

```suggestion
            roc_panic("list: is empty");
```